tests/app/questionnaire/test_placeholder_parser.py

```diff
@@ -1137,3 +1137,99 @@ def test_placeholder_parser_calculated_summary_dependencies_cache(
     placeholder_2 = placeholder_parser(placeholder_list=placeholder_list_2)
     assert placeholder_2["unit-total-playback"] == 11
     assert path_finder.called == 1
+
+
+def test_placeholder_dependencies_cache(mocker, mock_renderer):
```

Comment:
Worth adding some doc strings to explain what exactly this is testing. Would be good to add to the existing one also.

Reply:
I was expecting a comment around the technical details as I don't think what is there currently adds much more than the method name and asserts?. I was expecting an explanation of why this test should work and how. Such as the first placeholder list uses transform x which is one of the one that require the path. The second list also uses the same transform and answers from the same section as the first call. Therefore, we are checking, on the second call the cache is used which was created when the first call was made. Something like this, need cleaning up but you get the idea. It will make it easier to understand when coming back to this, as a lot of going on and it isn't obvious why the test is doing the things it does.

Reply:
Still outstanding?

Reply:
Oh i had updated it but I stashed it when going to another branch. Pushed 👍